### ISSUE TITLE
Building for atom-shell v0.21.0 fails

### ISSUE THREAD

**Issue by billt2006** [nodegit/nodegit#376]
If you try to build nodegit for atom-shell v0.21.0, it fails with the following messages. I have tried this under Mac OS X 10.9 and Arch Linux with the same results (v0.20.8 is fine). It seems like it could be an issue with nan?

```
{ [Error: Command failed: ../vendor/libgit2/src/attr.c:10:1: warning: unused function 'kh_clear_str' [-Wunused-function]
GIT__USE_STRMAP;
^
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:206:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_clear_##name(kh_##name##_t *h)                                            \
                   ^
<scratch space>:12:1: note: expanded from here
kh_clear_str
^
../vendor/libgit2/src/attr.c:10:1: warning: unused function 'kh_del_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:333:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_del_##name(kh_##name##_t *h, khint_t x)                           \
                   ^
<scratch space>:26:1: note: expanded from here
kh_del_str
^
2 warnings generated.
../vendor/libgit2/src/attrcache.c:8:1: warning: unused function 'kh_clear_str' [-Wunused-function]
GIT__USE_STRMAP;
^
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:206:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_clear_##name(kh_##name##_t *h)                                            \
                   ^
<scratch space>:12:1: note: expanded from here
kh_clear_str
^
../vendor/libgit2/src/attrcache.c:8:1: warning: unused function 'kh_del_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:333:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_del_##name(kh_##name##_t *h, khint_t x)                           \
                   ^
<scratch space>:26:1: note: expanded from here
kh_del_str
^
2 warnings generated.
../vendor/libgit2/src/clone.c:444:20: warning: unused function 'repository_base' [-Wunused-function]
static const char *repository_base(git_repository *repo)
                   ^
1 warning generated.
../vendor/libgit2/src/config_file.c:24:1: warning: unused function 'kh_clear_str' [-Wunused-function]
GIT__USE_STRMAP;
^
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:206:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_clear_##name(kh_##name##_t *h)                                            \
                   ^
<scratch space>:12:1: note: expanded from here
kh_clear_str
^
../vendor/libgit2/src/config_file.c:24:1: warning: unused function 'kh_del_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:333:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_del_##name(kh_##name##_t *h, khint_t x)                           \
                   ^
<scratch space>:26:1: note: expanded from here
kh_del_str
^
2 warnings generated.
../vendor/libgit2/src/diff_driver.c:20:1: warning: unused function 'kh_clear_str' [-Wunused-function]
GIT__USE_STRMAP;
^
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:206:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_clear_##name(kh_##name##_t *h)                                            \
                   ^
<scratch space>:12:1: note: expanded from here
kh_clear_str
^
../vendor/libgit2/src/diff_driver.c:20:1: warning: unused function 'kh_del_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:333:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_del_##name(kh_##name##_t *h, khint_t x)                           \
                   ^
<scratch space>:26:1: note: expanded from here
kh_del_str
^
2 warnings generated.
../vendor/libgit2/src/odb_mempack.c:21:1: warning: unused function 'kh_destroy_oid' [-Wunused-function]
GIT__USE_OIDMAP;
^
../vendor/libgit2/src/oidmap.h:30:2: note: expanded from macro 'GIT__USE_OIDMAP'
        __KHASH_IMPL(oid, static kh_inline, const git_oid *, void *, 1, git_oidmap_hash, git_oid_equal)
        ^
../vendor/libgit2/src/khash.h:198:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_destroy_##name(kh_##name##_t *h)                                          \
                   ^
<scratch space>:5:1: note: expanded from here
kh_destroy_oid
^
../vendor/libgit2/src/odb_mempack.c:21:1: warning: unused function 'kh_clear_oid' [-Wunused-function]
../vendor/libgit2/src/oidmap.h:30:2: note: expanded from macro 'GIT__USE_OIDMAP'
        __KHASH_IMPL(oid, static kh_inline, const git_oid *, void *, 1, git_oidmap_hash, git_oid_equal)
        ^
../vendor/libgit2/src/khash.h:206:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_clear_##name(kh_##name##_t *h)                                            \
                   ^
<scratch space>:8:1: note: expanded from here
kh_clear_oid
^
../vendor/libgit2/src/odb_mempack.c:21:1: warning: unused function 'kh_del_oid' [-Wunused-function]
../vendor/libgit2/src/oidmap.h:30:2: note: expanded from macro 'GIT__USE_OIDMAP'
        __KHASH_IMPL(oid, static kh_inline, const git_oid *, void *, 1, git_oidmap_hash, git_oid_equal)
        ^
../vendor/libgit2/src/khash.h:333:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_del_##name(kh_##name##_t *h, khint_t x)                           \
                   ^
<scratch space>:22:1: note: expanded from here
kh_del_oid
^
3 warnings generated.
../vendor/libgit2/src/refdb_fs.c:29:1: warning: unused function 'kh_init_str' [-Wunused-function]
GIT__USE_STRMAP;
^
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:195:23: note: expanded from macro '__KHASH_IMPL'
        SCOPE kh_##name##_t *kh_init_##name(void) {                                                     \
                             ^
<scratch space>:62:1: note: expanded from here
kh_init_str
^
../vendor/libgit2/src/refdb_fs.c:29:1: warning: unused function 'kh_destroy_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:198:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_destroy_##name(kh_##name##_t *h)                                          \
                   ^
<scratch space>:67:1: note: expanded from here
kh_destroy_str
^
../vendor/libgit2/src/refdb_fs.c:29:1: warning: unused function 'kh_clear_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:206:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_clear_##name(kh_##name##_t *h)                                            \
                   ^
<scratch space>:70:1: note: expanded from here
kh_clear_str
^
../vendor/libgit2/src/refdb_fs.c:29:1: warning: unused function 'kh_get_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:213:16: note: expanded from macro '__KHASH_IMPL'
        SCOPE khint_t kh_get_##name(const kh_##name##_t *h, khkey_t key)        \
                      ^
<scratch space>:73:1: note: expanded from here
kh_get_str
^
../vendor/libgit2/src/refdb_fs.c:29:1: warning: unused function 'kh_put_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:291:16: note: expanded from macro '__KHASH_IMPL'
        SCOPE khint_t kh_put_##name(kh_##name##_t *h, khkey_t key, int *ret) \
                      ^
<scratch space>:79:1: note: expanded from here
kh_put_str
^
../vendor/libgit2/src/refdb_fs.c:29:1: warning: unused function 'kh_del_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:333:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_del_##name(kh_##name##_t *h, khint_t x)                           \
                   ^
<scratch space>:84:1: note: expanded from here
kh_del_str
^
6 warnings generated.
../vendor/libgit2/src/refs.c:26:1: warning: unused function 'kh_init_str' [-Wunused-function]
GIT__USE_STRMAP;
^
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:195:23: note: expanded from macro '__KHASH_IMPL'
        SCOPE kh_##name##_t *kh_init_##name(void) {                                                     \
                             ^
<scratch space>:62:1: note: expanded from here
kh_init_str
^
../vendor/libgit2/src/refs.c:26:1: warning: unused function 'kh_destroy_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:198:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_destroy_##name(kh_##name##_t *h)                                          \
                   ^
<scratch space>:67:1: note: expanded from here
kh_destroy_str
^
../vendor/libgit2/src/refs.c:26:1: warning: unused function 'kh_clear_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:206:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_clear_##name(kh_##name##_t *h)                                            \
                   ^
<scratch space>:70:1: note: expanded from here
kh_clear_str
^
../vendor/libgit2/src/refs.c:26:1: warning: unused function 'kh_get_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:213:16: note: expanded from macro '__KHASH_IMPL'
        SCOPE khint_t kh_get_##name(const kh_##name##_t *h, khkey_t key)        \
                      ^
<scratch space>:73:1: note: expanded from here
kh_get_str
^
../vendor/libgit2/src/refs.c:26:1: warning: unused function 'kh_put_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:291:16: note: expanded from macro '__KHASH_IMPL'
        SCOPE khint_t kh_put_##name(kh_##name##_t *h, khkey_t key, int *ret) \
                      ^
<scratch space>:79:1: note: expanded from here
kh_put_str
^
../vendor/libgit2/src/refs.c:26:1: warning: unused function 'kh_del_str' [-Wunused-function]
../vendor/libgit2/src/strmap.h:23:2: note: expanded from macro 'GIT__USE_STRMAP'
        __KHASH_IMPL(str, static kh_inline, const char *, void *, 1, kh_str_hash_func, kh_str_hash_equal)
        ^
../vendor/libgit2/src/khash.h:333:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_del_##name(kh_##name##_t *h, khint_t x)                           \
                   ^
<scratch space>:84:1: note: expanded from here
kh_del_str
^
6 warnings generated.
../vendor/libgit2/src/submodule.c:86:1: warning: unused function 'kh_clear_str' [-Wunused-function]
__KHASH_IMPL(
^
../vendor/libgit2/src/khash.h:206:13: note: expanded from macro '__KHASH_IMPL'
        SCOPE void kh_clear_##name(kh_##name##_t *h)                                            \
                   ^
<scratch space>:12:1: note: expanded from here
kh_clear_str
^
1 warning generated.
../vendor/libssh2/src/misc.c:205:19: warning: unused variable 'base64_table' [-Wunused-const-variable]
static const char base64_table[] =
                  ^
../vendor/libssh2/src/misc.c:214:19: warning: unused variable 'base64_pad' [-Wunused-const-variable]
static const char base64_pad = '=';
                  ^
2 warnings generated.
../vendor/openssl/openssl/ssl/d1_srtp.c:147:7: warning: missing field 'id' initializer [-Wmissing-field-initializers]
    {0}
      ^
1 warning generated.
../vendor/openssl/openssl/crypto/asn1/x_bignum.c:85:1: warning: missing field 'prim_print' initializer [-Wmissing-field-initializers]
};
^
1 warning generated.
../vendor/openssl/openssl/crypto/cmac/cm_ameth.c:96:2: warning: missing field 'item_verify' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/dh/dh_ameth.c:500:2: warning: missing field 'old_priv_decode' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/dsa/dsa_ameth.c:650:3: warning: missing field 'pem_str' initializer [-Wmissing-field-initializers]
                },
                ^
../vendor/openssl/openssl/crypto/dsa/dsa_ameth.c:656:3: warning: missing field 'pem_str' initializer [-Wmissing-field-initializers]
                },
                ^
../vendor/openssl/openssl/crypto/dsa/dsa_ameth.c:662:3: warning: missing field 'pem_str' initializer [-Wmissing-field-initializers]
                },
                ^
../vendor/openssl/openssl/crypto/dsa/dsa_ameth.c:668:3: warning: missing field 'pem_str' initializer [-Wmissing-field-initializers]
                },
                ^
../vendor/openssl/openssl/crypto/dsa/dsa_ameth.c:702:3: warning: missing field 'item_verify' initializer [-Wmissing-field-initializers]
                }
                ^
5 warnings generated.
../vendor/openssl/openssl/crypto/ec/ec_ameth.c:660:2: warning: missing field 'item_verify' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/ec/ec_lib.c:71:19: warning: unused variable 'EC_version' [-Wunused-const-variable]
static const char EC_version[] = "EC" OPENSSL_VERSION_PTEXT;
                  ^
1 warning generated.
../vendor/openssl/openssl/crypto/engine/eng_dyn.c:117:30: warning: unused variable 'dynamic_cmd_defns_empty' [-Wunused-const-variable]
static const ENGINE_CMD_DEFN dynamic_cmd_defns_empty[] = {
                             ^
1 warning generated.
../vendor/openssl/openssl/crypto/engine/eng_openssl.c:344:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_dss.c:94:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_dss1.c:96:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_ecdsa.c:144:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_md4.c:97:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_md5.c:96:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_mdc2.c:97:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_null.c:88:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_ripemd.c:96:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_sha.c:95:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/evp/m_sha1.c:97:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
../vendor/openssl/openssl/crypto/evp/m_sha1.c:134:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
../vendor/openssl/openssl/crypto/evp/m_sha1.c:153:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
../vendor/openssl/openssl/crypto/evp/m_sha1.c:184:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
../vendor/openssl/openssl/crypto/evp/m_sha1.c:203:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
5 warnings generated.
../vendor/openssl/openssl/crypto/evp/m_wp.c:37:2: warning: missing field 'md_ctrl' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/hmac/hm_ameth.c:166:2: warning: missing field 'item_verify' initializer [-Wmissing-field-initializers]
        };
        ^
1 warning generated.
../vendor/openssl/openssl/crypto/pem/pem_lib.c:473:6: warning: variable 'i' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
        if (o)
            ^
../vendor/openssl/openssl/crypto/pem/pem_lib.c:480:5: note: uninitialized use occurs here
        j+=i;
           ^
../vendor/openssl/openssl/crypto/pem/pem_lib.c:473:2: note: remove the 'if' if its condition is always true
        if (o)
        ^~~~~~
../vendor/openssl/openssl/crypto/pem/pem_lib.c:443:7: note: initialize the variable 'i' to silence this warning
        int i,j,o,klen;
             ^
              = 0
1 warning generated.
../vendor/openssl/openssl/crypto/rsa/rsa_ameth.c:697:3: warning: missing field 'pem_str' initializer [-Wmissing-field-initializers]
                }
                ^
1 warning generated.
In file included from ../src/nodegit.cc:6:
In file included from ../src/../include/wrapper.h:11:
../node_modules/nan/nan.h:67:21: error: no type named 'ExternalAsciiStringResource' in 'v8::String'; did you mean 'ExternalStringResource'?
typedef v8::String::ExternalAsciiStringResource
        ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
                    ExternalStringResource
/Users/billthornton/.atom-shell-gyp/.node-gyp/0.21.0/deps/v8/include/v8.h:1936:19: note: 'ExternalStringResource' declared here
  class V8_EXPORT ExternalStringResource
                  ^
In file included from ../src/nodegit.cc:6:
In file included from ../src/../include/wrapper.h:11:
In file included from ../node_modules/nan/nan.h:74:
../node_modules/nan/nan_new.h:155:26: error: class member cannot be redeclared
  static inline return_t New(NanExternalOneByteStringResource * value);
                         ^
../node_modules/nan/nan_new.h:154:26: note: previous declaration is here
  static inline return_t New(v8::String::ExternalStringResource * value);
                         ^
In file included from ../src/nodegit.cc:6:
In file included from ../src/../include/wrapper.h:11:
In file included from ../node_modules/nan/nan.h:74:
In file included from ../node_modules/nan/nan_new.h:181:
../node_modules/nan/nan_implementation_12_inl.h:213:22: error: redefinition of 'New'
Factory<v8::String>::New(NanExternalOneByteStringResource * value) {
                     ^
../node_modules/nan/nan_implementation_12_inl.h:208:22: note: previous definition is here
Factory<v8::String>::New(v8::String::ExternalStringResource * value) {
                     ^
In file included from ../src/nodegit.cc:6:
In file included from ../src/../include/wrapper.h:11:
In file included from ../node_modules/nan/nan.h:74:
../node_modules/nan/nan_new.h:310:1: error: redefinition of 'NanNew'
NanNew(NanExternalOneByteStringResource * value) {
^
../node_modules/nan/nan_new.h:304:1: note: previous definition is here
NanNew(v8::String::ExternalStringResource * value) {
^
In file included from ../src/nodegit.cc:6:
In file included from ../src/../include/wrapper.h:11:
../node_modules/nan/nan.h:1733:17: error: no member named 'GetExternalAsciiStringResource' in 'v8::String'; did you mean 'GetExternalOneByteStringResource'?
    return str->GetExternalAsciiStringResource();
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                GetExternalOneByteStringResource
/Users/billthornton/.atom-shell-gyp/.node-gyp/0.21.0/deps/v8/include/v8.h:2003:40: note: 'GetExternalOneByteStringResource' declared here
  const ExternalOneByteStringResource* GetExternalOneByteStringResource() const;
                                       ^
In file included from ../src/nodegit.cc:6:
In file included from ../src/../include/wrapper.h:11:
../node_modules/nan/nan.h:1733:12: error: cannot initialize return object of type 'const NanExternalOneByteStringResource *' (aka 'const v8::String::ExternalStringResource *') with an rvalue of type 'const v8::String::ExternalOneByteStringResource *'
    return str->GetExternalAsciiStringResource();
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:1743:17: error: no member named 'IsExternalAscii' in 'v8::String'; did you mean 'IsExternal'?
    return str->IsExternalAscii();
                ^~~~~~~~~~~~~~~
                IsExternal
/Users/billthornton/.atom-shell-gyp/.node-gyp/0.21.0/deps/v8/include/v8.h:1900:8: note: 'IsExternal' declared here
  bool IsExternal() const;
       ^
In file included from ../src/nodegit.cc:6:
In file included from ../src/../include/wrapper.h:11:
../node_modules/nan/nan.h:1768:11: error: assigning to 'const char *' from incompatible type 'const uint16_t *' (aka 'const unsigned short *')
    *data = ext->data();
          ^ ~~~~~~~~~~~
8 errors generated.
make: *** [Release/obj.target/nodegit/src/nodegit.o] Error 1
```

**Comment by maxkorp** [comment]
What command are you using to build? Are you running apm install? npm install with a parent dir with an atom-shell engine?

**Comment by billt2006** [comment]
I was using npm install with a parent dir with an atom-shell engine.

**Comment by johnhaley81** [comment]
Just a quick note that is is in 0.21.0 and NOT 0.20.8 of atom-shell.

**Comment by billt2006** [comment]
That's right. 0.20.8 is good. 0.21.0 fails.

**Comment by maxkorp** [comment]
Weird, ok, I'll take a crack at it.

**Comment by maxkorp** [comment]
I get no issues with 0.20.8 or 0.21.0

**Comment by billt2006** [comment]
I don't see anything weird in it. I wiped out ~/.atom-shell-gyp and tried to build again and it still failed.

**Comment by maxkorp** [comment]
Can you list the contents of `~/.atom-shell-gyp/.node-gyp/0.21.0` (or whichever version you try to build against)?

**Comment by billt2006** [comment]
```
 ~/.atom-shell-gyp/.node-gyp/0.21.0
├── common.gypi
├── config.gypi
├── deps
│   ├── http_parser
│   │   └── http_parser.h
│   ├── mdb_v8
│   │   ├── v8cfg.h
│   │   └── v8dbg.h
│   ├── npm
│   │   └── node_modules
│   │       └── node-gyp
│   │           └── addon.gypi
│   ├── uv
│   │   ├── common.gypi
│   │   ├── include
│   │   │   ├── android-ifaddrs.h
│   │   │   ├── pthread-fixes.h
│   │   │   ├── stdint-msvc2008.h
│   │   │   ├── tree.h
│   │   │   ├── uv-aix.h
│   │   │   ├── uv-bsd.h
│   │   │   ├── uv-darwin.h
│   │   │   ├── uv-errno.h
│   │   │   ├── uv-linux.h
│   │   │   ├── uv-sunos.h
│   │   │   ├── uv-threadpool.h
│   │   │   ├── uv-unix.h
│   │   │   ├── uv-version.h
│   │   │   ├── uv-win.h
│   │   │   └── uv.h
│   │   ├── samples
│   │   │   └── socks5-proxy
│   │   │       ├── defs.h
│   │   │       └── s5.h
│   │   ├── src
│   │   │   ├── heap-inl.h
│   │   │   ├── queue.h
│   │   │   ├── unix
│   │   │   │   ├── atomic-ops.h
│   │   │   │   ├── internal.h
│   │   │   │   ├── linux-syscalls.h
│   │   │   │   └── spinlock.h
│   │   │   ├── uv-common.h
│   │   │   └── win
│   │   │       ├── atomicops-inl.h
│   │   │       ├── handle-inl.h
│   │   │       ├── internal.h
│   │   │       ├── req-inl.h
│   │   │       ├── stream-inl.h
│   │   │       ├── winapi.h
│   │   │       └── winsock.h
│   │   └── test
│   │       ├── benchmark-list.h
│   │       ├── runner-unix.h
│   │       ├── runner-win.h
│   │       ├── runner.h
│   │       ├── task.h
│   │       └── test-list.h
│   ├── v8
│   │   └── include
│   │       ├── libplatform
│   │       │   └── libplatform.h
│   │       ├── v8-debug.h
│   │       ├── v8-platform.h
│   │       ├── v8-profiler.h
│   │       ├── v8-testing.h
│   │       ├── v8-util.h
│   │       ├── v8.h
│   │       └── v8config.h
│   └── zlib
│       ├── contrib
│       │   ├── blast
│       │   │   └── blast.h
│       │   ├── infback9
│       │   │   ├── infback9.h
│       │   │   ├── inffix9.h
│       │   │   ├── inflate9.h
│       │   │   └── inftree9.h
│       │   ├── iostream
│       │   │   └── zfstream.h
│       │   ├── iostream2
│       │   │   └── zstream.h
│       │   ├── iostream3
│       │   │   └── zfstream.h
│       │   ├── minizip
│       │   │   ├── crypt.h
│       │   │   ├── ioapi.h
│       │   │   ├── iowin32.h
│       │   │   ├── mztools.h
│       │   │   ├── unzip.h
│       │   │   └── zip.h
│       │   └── puff
│       │       └── puff.h
│       ├── crc32.h
│       ├── deflate.h
│       ├── examples
│       │   └── gzlog.h
│       ├── gzguts.h
│       ├── inffast.h
│       ├── inffixed.h
│       ├── inflate.h
│       ├── inftrees.h
│       ├── trees.h
│       ├── zconf.h
│       ├── zlib.h
│       └── zutil.h
├── installVersion
└── src
    ├── async-wrap-inl.h
    ├── async-wrap.h
    ├── base-object-inl.h
    ├── base-object.h
    ├── debug-agent.h
    ├── env-inl.h
    ├── env.h
    ├── handle_wrap.h
    ├── node.h
    ├── node_buffer.h
    ├── node_constants.h
    ├── node_counters.h
    ├── node_crypto.h
    ├── node_crypto_bio.h
    ├── node_crypto_clienthello-inl.h
    ├── node_crypto_clienthello.h
    ├── node_crypto_groups.h
    ├── node_dtrace.h
    ├── node_file.h
    ├── node_http_parser.h
    ├── node_i18n.h
    ├── node_internals.h
    ├── node_javascript.h
    ├── node_object_wrap.h
    ├── node_root_certs.h
    ├── node_stat_watcher.h
    ├── node_v8_platform.h
    ├── node_version.h
    ├── node_watchdog.h
    ├── node_win32_etw_provider-inl.h
    ├── node_win32_etw_provider.h
    ├── node_win32_perfctr_provider.h
    ├── node_wrap.h
    ├── pipe_wrap.h
    ├── queue.h
    ├── req_wrap.h
    ├── smalloc.h
    ├── spawn_sync.h
    ├── stream_wrap.h
    ├── string_bytes.h
    ├── tcp_wrap.h
    ├── tls_wrap.h
    ├── tree.h
    ├── tty_wrap.h
    ├── udp_wrap.h
    ├── util-inl.h
    ├── util.h
    └── v8abbr.h

28 directories, 128 files
```

**Comment by maxkorp** [comment]
Man, that seems right. Can I see your package.json?

**Comment by billt2006** [comment]
There isn't much to it...
```json
{
  "name": "name",
  "displayName": "Name",
  "version": "0.0.0",
  "description": "",
  "keywords": [],
  "main": "main.js",
  "dependencies": {},
  "devDependencies": {
    "concat-files": "~0.1.0",
    "grunt": "~0.4.5",
    "grunt-build-atom-shell": "^1.1.5",
    "grunt-contrib-clean": "~0.6.0",
    "grunt-contrib-compress": "^0.10.0",
    "grunt-contrib-concat": "~0.5.0",
    "grunt-contrib-copy": "~0.5.0",
    "grunt-contrib-jshint": "~0.10.0",
    "grunt-contrib-rename": "0.0.3",
    "grunt-contrib-watch": "^0.6.1",
    "grunt-download-atom-shell": "^0.11.0",
    "load-grunt-tasks": "^0.6.0",
    "time-grunt": "^0.4.0"
  },
  "engines": {
    "node": ">=0.8.0",
    "atom-shell": "0.21.0"
  }
}
```

**Comment by maxkorp** [comment]
Ha, found it!
NAN checks the node version to compare against v10/11, and it's clashing with the way atom-shell versions node.

**Comment by maxkorp** [comment]
https://github.com/rvagg/nan/issues/226#issuecomment-71140076

**Comment by tbranyen** [comment]
Resolved via f0b29f5873161bd5c6c53896929ee7724a090427 thanks @orderedlist !
